client/alloc_endpoint.go

```diff
@@ -343,3 +375,49 @@ func (s *execStream) Recv() (*drivers.ExecTaskStreamingRequestMsg, error) {
 	err := s.decoder.Decode(&req)
 	return &req, err
 }
+
+func validateActionExists(action, task, taskGroup string, alloc *nstructs.Allocation) (*nstructs.Action, error) {
+	if taskGroup == "" {
+		return nil, fmt.Errorf("Must supply a task group")
+	}
+
+	if task == "" {
+		return nil, fmt.Errorf("Must supply a task")
+	}
+
+	var tg *nstructs.TaskGroup
+	for _, group := range alloc.Job.TaskGroups {
+		if group.Name == taskGroup {
+			tg = group
+			break
+		}
+	}
+	if tg == nil {
+		return nil, fmt.Errorf("task group %s not found", taskGroup)
+	}
```

Comment:
An `Allocation` always belongs to exactly one task group, so there's a `TaskGroup` field with the name of the task group. You can then use the `LookupTaskGroup` helper on `structs.Job` to get the `*structs.TaskGroup` you need here.